main.cpp

```diff
@@ -1722,6 +1724,38 @@ void loadAllowedLateDays(std::vector<Student*> &students) {
   }
 }
 
+void SaveExtensionReports(const std::vector<Student*> &students) {
+  system ("mkdir -p student_extension_reports");
+  for (size_t i=0;i<students.size();i++) {
+    std::vector<std::tuple<ItemGrade,std::tuple<GRADEABLE_ENUM,int> > > gradeablesWithExtensions = students[i]->getItemsWithExceptions();
+    std::string username = students[i]->getUserName();
+    std::ofstream student_ostr("student_extension_reports/"+username+".html");
+    assert (student_ostr.good());
+    
+    student_ostr << "<h3> Excused Absence Extensions for: " << username << "</h3>" << std::endl;
+    student_ostr << "<table cellpadding=5 style=\"border:1px solid #aaaaaa; background-color:#ffffff;\">" << std::endl;
+    student_ostr << "<tr><td>Gradeable</td><td align=center>Days Extended</td><td align=center>Reason</td><td></td></tr>" << std::endl;
+    if (gradeablesWithExtensions.size() == 0) {
+      student_ostr << "<tr><td>N/a</td></tr>" << std::endl;
```

Comment:
Rainbowgrades should only display things that are relevant  to the student. Instead of having the table for all studennts and leaving it blank with N/a, only create/make it visible to students who actually got extension?

Reply:
Yeah sure, just changed it so that table is only displayed if the student has extensions.